meta-store/src/meta_store.rs

```diff
@@ -0,0 +1,163 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+mod single_file;
+
+use std::fmt::{Debug, Display};
+use std::io;
+use std::ops::Range;
+
+use async_trait::async_trait;
+use thiserror::Error;
+
+pub type IndexId = String;
+pub type SplitId = String;
+
+pub enum State {
```

Comment:
@fulmicoton @mosuka @guilload  can we be more specific as In `SplitState` ?

Reply:
@evanxg852000 @fulmicoton @guilload 
Yes, I think `SplitState` is better than `State`. Shall I rename it?

Reply:
I agree for SplitState